### ISSUE TITLE
Allow propertyMap rules at transformQuery

### ISSUE THREAD

**Issue by jnavb** [react-querybuilder/react-querybuilder#574]
the documentation of `ruleGroupProcessor` argument for `transformQuery` states:

```
    /**
     * When a group is encountered in the hierarchy, it will be replaced
     * with the result of this function. Note that the `rules` property from
     * the original group will be processed as normal and reapplied to the
     * new group object.
     *
     * @defaultValue `rg => rg`
     */
```

is there a reason why a change in the property rules is not supported?

**Comment by jakeboone02** [comment]
You know, I think maybe it was just easier to do it that way when I first implemented `transformQuery`. As of now I can't really see a reason to keep that limitation, so I'll see if I can remove it.

**Comment by jakeboone02** [comment]
@jnavb This has been resolved in [v6.5.3](https://github.com/react-querybuilder/react-querybuilder/releases/tag/v6.5.3), but I just realized I forgot to update the JSDoc in the code itself so it still says the `rules` property will always be retained even though that's not true. I'll fix that in the next release.

**Comment by jnavb** [comment]
@jakeboone02 just test it and works nicely, thanks!
